### ISSUE TITLE
Marketplace - Some weird interaction that goes on w/ the Featured agent cards (cards get cut off on sides)

### ISSUE THREAD

**Issue by ograce1421** [Significant-Gravitas/AutoGPT#8973]
### Describe your issue.

Some weird interaction that goes on w/ the Featured agent cards that I've noticed. 
When you navigate through the cards, it gets cut off on the sides.. Is it possible to make sure that the cards don't get cut off when a user is interacting with the cards? and also is it possible to make the cards expand with the window?
this awkward gap here
![image](https://github.com/user-attachments/assets/b6c0783c-07ca-4bb1-a1f0-47885661f2a1)

Can we model it like these cards that Apple has on their app store website?
Like these cards here
https://github.com/user-attachments/assets/9e02b750-5002-40e1-8a9e-3dd4f9a0883a

They're at the very bottom of this page: https://www.apple.com/app-store/
Basically these cards are exactly what I was envisioning for our cards
Take a look on how they animate, how the buttons states are (they're grey if it's the end of the list), if you expand or compress the window, the cards are still in frame
